source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -145,6 +145,20 @@ bool AggressiveDCEPass::AllExtensionsSupported() const {
     if (extensions_allowlist_.find(extName) == extensions_allowlist_.end())
       return false;
   }
+  // only allow NonSemantic.Shader.DebugInfo.100, we cannot safely optimise
+  // around unknown extended
+  // instruction sets even if they are non-semantic
+  for (auto& inst : context()->module()->ext_inst_imports()) {
+    assert(inst.opcode() == SpvOpExtInstImport &&
+           "Expecting an import of an extension's instruction set.");
+    const char* extension_name =
+        reinterpret_cast<const char*>(&inst.GetInOperand(0).words[0]);
+    if (0 == std::strncmp(extension_name, "NonSemantic.", 12) &&
+        0 != std::strncmp(extension_name, "NonSemantic.Shader.DebugInfo.100",
+                          32)) {
+      return false;
```

Comment:
Actually, OpenCL.DebugInfo.100 **is** allowed in ADCE, at least it is not disallowed here. The only thing that is being disallowed by this logic is any NonSemantic instruction sets that are **not** Shader.DebugInfo.100. This makes sense since we cannot safely assume in general that "dead" NonSemantic instructions are removable.

Reply:
Rather, we might want to ask ourselves: do we want to create an "allow list" of Extended Instruction Sets in the same way we have an "allow list" for Extensions? In the past this has not been necessary since before OpenCL.DebugInfo, extended instructions did not have side effects and "dead" instructions could be deleted. But now with OpenCL.DebugInfo and NonSemantic.Shader.DebugInfo we cannot make this assumption, and we need to add special logic to correctly handle these extended instruction sets.

I think we are currently handling all extended instruction sets correctly, so logic to disallow unhandled extended instruction sets is not immediately needed, but it is something we might want to consider to make it easier to add extended instruction sets in the future.